packages/e2e-test-utils-playwright/src/editor/get-blocks.ts

```diff
@@ -3,25 +3,49 @@
  */
 import type { Editor } from './index';
 
+type Block = {
+	name: string;
+	attributes: Record< string, unknown >;
+	innerBlocks: Block[];
+};
+
 /**
  * Returns the edited blocks.
  *
  * @param this
+ * @param options
+ * @param options.full Whether to return the full block data or just the name and attributes.
  *
  * @return  The blocks.
  */
-export async function getBlocks( this: Editor ) {
-	return await this.page.evaluate( () => {
-		const blocks = window.wp.data.select( 'core/block-editor' ).getBlocks();
+export async function getBlocks( this: Editor, { full = false } = {} ) {
+	return await this.page.evaluate(
+		( [ _full ] ) => {
+			// Remove other unpredictable properties like clientId from blocks for testing purposes.
+			function recursivelyTransformBlocks( blocks: Block[] ): Block[] {
+				return blocks.map( ( block ) => ( {
+					name: block.name,
+					attributes: block.attributes,
+					innerBlocks: recursivelyTransformBlocks(
+						block.innerBlocks
+					),
+				} ) );
+			}
```

Comment:
Nit (non-blocking): This helper can be defined at the file level.

Reply:
Will it work given that playwright needs to inject the function out of scope into the browser context?

Reply:
Right. We can pass it to the `page.evaluate` as an arg, but probably not worth the hassle.